src/commands/clone.js

```diff
@@ -29,48 +32,58 @@ export async function pullExtensions(appId, destinationDir) {
   const installations = await appManager.getInstallations(appId);
   const n = installations.length;
   let i = 0;
-  for(const inst of installations) {
+  for (const inst of installations) {
     i++;
-    await spinify(pullExtension(destinationDir, inst), `Downloading extension ${i}/${n}: ${inst.canonicalName}...`);
+    await spinify(
+      pullExtension(destinationDir, inst),
+      `Downloading extension ${i}/${n}: ${inst.canonicalName}...`
+    );
   }
 }
 
 async function pullExtension(destinationDir, { extension, canonicalName }) {
   let pullError = {};
-  let pullSuccessful = false;
-  for (let i = 0; !pullSuccessful || i < 4; i++) {
+  for (let i = 0; i < 4; i++) {
     try {
       const url = await getExtensionUrl(extension);
       const tgzDir = (await tmp.dir()).path;
       await downloadFile(url, { directory: tgzDir, filename: 'extension.tgz' });
-      await shoutemUnpack(path.join(tgzDir, 'extension.tgz'), path.join(destinationDir, canonicalName));
-      pullSuccessful = true;
+      await shoutemUnpack(
+        path.join(tgzDir, 'extension.tgz'),
+        path.join(destinationDir, canonicalName)
+      );
+
+      return;
     } catch (error) {
-      pullError = error;
-      console.log(`Failed to download extension ${canonicalName}, try ${i + 1}/5`);
+      if (error.code !== 'ENOTEMPTY') {
+        pullError = error;
+      }
     }
   }
 
-  if (!pullSuccessful) {
+  if (!_.isEmpty(pullError)) {
```

Comment:
Correct me if I'm wrong but I think we have a problem.

Iteration 1: There's an error. Ln `pullError = error` is executed. 
Iteration 2: Success. But, pullError still contains the error from Iteration 1. We will throw the error.